fem/coefficient.hpp

```diff
@@ -624,47 +766,66 @@ class MatrixFunctionCoefficient : public MatrixCoefficient
       mat.SetSize(0);
    }
 
+   /// Evaluate the matrix coefficient at @a ip.
    virtual void Eval(DenseMatrix &K, ElementTransformation &T,
                      const IntegrationPoint &ip);
 
    virtual ~MatrixFunctionCoefficient() { }
 };
 
+
+
+/** @brief Matrix coefficient defined by an matrix of scalar coefficients.
+    Coefficients that are not set will evaluate to zero in the vector.  The
+    of coefficient is stored as a flat Array with indexing (i,j) -> i*width+j.
+    This object takes ownership of the array of coefficients inside it and
+    deletes them at object destruction.
+    */
 class MatrixArrayCoefficient : public MatrixCoefficient
 {
 private:
    Array<Coefficient *> Coeff;
    Array<bool> ownCoeff;
 
 public:
-
+   /** @brief Construct matrix of dim = height*width coefficients.
```

Comment:
The size here is incorrect. The matrix is a square matrix with size `dim*dim = height*width`.